client/signup/steps/woocommerce-install/step-store-address/index.tsx

```diff
@@ -48,7 +49,8 @@ export default function StepStoreAddress( props: WooCommerceStoreAddressProps )
 
 	const siteId = useSelector( getSelectedSiteId ) as number;
 
-	const countriesList = useSelector( ( state ) => getCountries( state, 'woocommerce' ) ) || [];
+	const countriesList =
+		useSelector( ( state: IAppState ) => getCountries( state, 'woocommerce' ) ) || {};
```

Comment:
Can we be certain that this default value change is going to be necessary only for the `woocommerce` subtree?

Reply:
I think so. The other subtrees are currently all arrays. This default value should always have been an object but from the history it looks like when someone changed this subtree from `'payments'` to `'woocommerce'` (and thus from an array to an object) they forgot to change its default value.

To echo some of my other comments: in practice `getCountries` should never be able to return a falsy value for a known subtree. Its initializers always specify a default value. So this default (and the `null` default in the selector) is mostly academic and would probably apply only if the data returned by the API was falsy (there are no types or guards on the functions which fetch data from the API to this tree which is not great).